data-prepper-api/src/test/java/com/amazon/dataprepper/model/event/JacksonEventImplTest.java

```diff
@@ -0,0 +1,348 @@
+        final TestObject value = new TestObject(nestedValue);
+        final String nestedKey = "foo.bar.field1";
+
+        event.put(key, value);
+        final String actualNestedValue = event.get(nestedKey, String.class);
+
+        assertThat(actualNestedValue, is(notNullValue()));
+        assertThat(actualNestedValue, is(equalTo(nestedValue)));
+
+        final String replacementValue = UUID.randomUUID().toString();
+        event.put(nestedKey, replacementValue);
+        final TestObject result = event.get(key, TestObject.class);
+
+        assertThat(result, is(notNullValue()));
+        assertThat(result.getField1(), is(equalTo(replacementValue)));
+    }
+
+    @Test
+    public void testGet_withIncorrectPojo() {
+        final String key = "foo.bar";
+        final String nestedValue = UUID.randomUUID().toString();
+        final TestObject value = new TestObject(nestedValue);
+
+        event.put(key, value);
+
+        assertThrows(RuntimeException.class, () -> event.get(key, UUID.class));
+    }
+
+    @Test
+    public void testGet_withEmptyEvent() {
+        final String key = "foo.bar";
+
+        UUID result = event.get(key, UUID.class);
+
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void testOverwritingExistingKey() {
+        final String key = "foo.bar";
+        final UUID value = UUID.randomUUID();
+
+        event.put(key, UUID.randomUUID());
+        event.put(key, value);
+        final UUID result = event.get(key, UUID.class);
+
+        assertThat(result, is(notNullValue()));
+        assertThat(result, is(equalTo(value)));
+    }
+
+    @Test
+    public void testDeletingKey() {
+        final String key = "foo";
+
+        event.put(key, UUID.randomUUID());
+        event.delete(key);
+        final UUID result = event.get(key, UUID.class);
+
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void testDelete_withNestedKey() {
+        final String key = "foo.bar";
+
+        event.put(key, UUID.randomUUID());
+        event.delete(key);
+        final UUID result = event.get(key, UUID.class);
+
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void testDelete_withNonexistentKey() {
+        final String key = "foo.bar";
+        UUID result = event.get(key, UUID.class);
+        assertThat(result, is(nullValue()));
+
+        event.delete(key);
+
+        result = event.get(key, UUID.class);
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void testKey_withNull_throwsNullPointerException() {
+        assertThrowsForKeyCheck(NullPointerException.class, null);
+    }
+
+    @Test
+    public void testKey_withString_throwsException() {
+        assertThrowsForKeyCheck(IllegalArgumentException.class, "");
+    }
+
+    @Test
+    public void testKey_withSpecialCharacters_throwsException() {
+        assertThrowsForKeyCheck(IllegalArgumentException.class, "bogusKey&");
+    }
+
+    @Test
```

Comment:
You can drop the `assertThrowsForKeyCheck` and consolidate the code using JUnit 5's `@ParameterizedTest`.

It would look something like this:

```
@ParameterizedTest
@ValueSource(strings = ["", "bogusKey&" /* ... others */])
void testKey_withBadInput_throwsException(String badKey) {
         assertThrows(expectedThrowable, () -> event.put(badKey, UUID.randomUUID()));
         assertThrows(expectedThrowable, () -> event.get(badKey, String.class));
         assertThrows(expectedThrowable, () -> event.delete(badKey));
}
```

It will not work for `null`. You will still need its own test.

Reply:
I thought about doing this but decided against it because I could make a descriptive test name for each scenario. I can adopt the proposal if this is preferred